test/deriver/errors/factory_abstract_type.ml

```diff
@@ -0,0 +1,2 @@
+type t = int
```

Comment:
I am probably just misunderstanding the type system in some way, but why is this type abstract? I thought that abstract types can only be of the form:

```ocaml
type t
```

whereas in this case the type is concretely an int.

Reply:
It is indeed a bit misleading as such type definitions are called `abstract` in the AST types. I think you should think of them as defined in terms of other existing types instead of explicitly defining their structure.

The reason why they are rejected here is because `[@@deriving factory]` only supports variants and record types.

Reply:
That makes sense thanks